### PR TITLE
FF-4241: Upgrade jetty version for 5.3.x and earlier

### PR DESCRIPTION
**Pull request by xjin-Confluent**
https://confluentinc.atlassian.net/browse/FF-4241

### CODE REVIEW

**Comment by omkreddy** [comment]
Note that we may need to fix compilation/test failures due to the change. I think we need to run platform system tests with this change + kafka changes.

File: pom.xml [hunk 1]
```diff
@@ -55,7 +55,7 @@
         <!-- can't upgrade jersey past 2.25 until we move to java 8 -->
         <jersey.version>2.25</jersey.version>
```
Comment (omkreddy) [inline pom.xml:56]: maybe we can also update jersey to 2.31

File: pom.xml [hunk 2]
```diff
@@ -55,7 +55,7 @@
         <!-- can't upgrade jersey past 2.25 until we move to java 8 -->
```
Comment (omkreddy) [inline pom.xml:55]: its better to merge this PR from 5.0 branch on-wards.

**Comment by omkreddy** [comment]
its better to merge this PR from 5.0 branch on-wards.

File: pom.xml (hunk 1 above)
Comment (xjin-Confluent) [reply pom.xml:56]: noticed here: https://confluentinc.atlassian.net/browse/CPKAFKA-6127. Will add that change.

**Comment by xjin-Confluent** [comment]
> its better to merge this PR from 5.0 branch on-wards.

 Yep, it's better skipping 4.0.x. I will change my base branch.

**Comment by omkreddy** [comment]
looks like SSL tests are failing. maybe we need to backport some changes from 5.4 branches.

this may be risky update. maybe we should consider updating jetty from 5.4.x and above.

**Comment by omkreddy** [comment]
Also looks like we have dependency on jersey 2.27 and below for some modules like ksql/control center. So we can not update jersey without code changes to those modules :(.   I think, we can not update jersey for 5.3.x and earlier.

**Pull request closed without merging**